### PR TITLE
Update CJ example for sorted=FALSE

### LINKED ISSUE #6215: Update documentation for CJ supporting complex number
In the documentation, pg no. 73, it says the function CJ  "support for sorting complex is not yet implemented".
I expected to get a warning, but the sorting looks correct. It looks like it has been added in the pull request https://github.com/Rdatatable/data.table/issues/3690.  If the functionality is implemented, would it be possible to update the documentation to remove this comment: "support for sorting complex is not yet implemented"

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #6215

### CODE REVIEW

**Review by Anirban166 — APPROVED** [review]

File: man/J.Rd [hunk 1]
```diff
@@ -56,9 +56,7 @@ x = c(1, 1, 2)
 y = c(4, 6, 4)
 CJ(x, y)              # output columns are automatically named 'x' and 'y'
 CJ(x, y, unique=TRUE) # unique(x) and unique(y) are computed automatically
-
-z = 0:1 + (0:1)*1i
-CJ(x, z, sorted = FALSE) # support for sorting complex is not yet implemented
+CJ(x, y, sorted = FALSE) # retain input order for y
```
[the remark this answers is not in the document]
Comment (Anirban166) [reply man/J.Rd:59]: (Following the examples above such as `setkey(DT, B)   # reorders table and marks it sorted` and
`DT[J("b")]      # returns the 2nd row` for consistency)

[the remark this answers is not in the document]
Comment (MichaelChirico) [reply man/J.Rd:59]: I think both work, I prefer 'retain'

[the remark this answers is not in the document]
Comment (Anirban166) [reply man/J.Rd:59]: Works!

[the remark this answers is not in the document]
Comment (MichaelChirico) [reply man/J.Rd:59]: Merged before your follow-up comment. I had only looked at the previous two comments to check consistency. Feel free to do a follow-up PR :)

**Pull request merged**